### PR TITLE
Migrate to vtex.css-handles@1.x, migrate to TypeScript and deprecate some components

### PR DESCRIPTION
**Pull request by klzns**
<br/>
<br/>

### Tip for reviewers: review commit by commit (I put an effort on this ❤️ )

<br/>
<br/>

#### What problem is this solving?

Make it possible to customize component's CSS Handles via JavaScript.

#### How to test it?

https://summary--storecomponents.myvtex.com/
https://brenovtex--carrefourbrfood.myvtex.com/Alimentos-e-Bebidas/Bebidas?crfimt=hm-tlink|mercado|menu|bebidas|2|091220
https://brenovtex--muji.myvtex.com/furniture?__bindingAddress=www.mujionline.eu/uk
https://brenovtex--alssports.myvtex.com/jacket?_q=jacket&map=ft

#### Screenshots or example usage:

n/a

#### Describe alternatives you've considered, if any.

n/a

#### Related to / Depends on

Depends on: 
1. https://github.com/vtex-apps/product-summary-context/pull/16
2. https://github.com/vtex-apps/responsive-values/pull/10

### CODE REVIEW

File: react/ProductSummarySKUSelector.tsx [hunk 1]
```diff
@@ -30,12 +34,15 @@ function ProductSummarySKUSelector(props: any) {
...
-      product.items && product.items.find((item: any) => item.itemId === skuId)
+      product.items &&
+      (product.items.find(
+        (item) => item.itemId === skuId
+      ) as ProductSummaryTypes.SKU)
 
     const sku = {
       ...selectedItem,
-      image: head(selectedItem.images),
-      seller: head(selectedItem.sellers),
```
Comment (kaisermann) [inline react/ProductSummarySKUSelector.tsx:38]: AAAAAAAAAAAAAAA

File: react/hooks/useSetProduct.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    productDispatch?.({
+      type: 'SET_PRODUCT',
+      args: { product: (product as unknown) as ProductTypes.Product },
+    })
+
+    productDispatch?.({
+      type: 'SET_SELECTED_ITEM',
+      args: { item: (newSelectedItem as unknown) as ProductTypes.Item },
+    })
+  }
```
Comment (kaisermann) [inline react/hooks/useSetProduct.ts:32]: glorious

File: react/hooks/useSimulation.ts [hunk 3]
```diff
@@ -1,10 +1,15 @@
 import { useMemo } from 'react'
 import { useQuery } from 'react-apollo'
-import itemsWithSimulationQuery from 'vtex.store-resources/QueryItemsWithSimulation'
+import type { ProductSummaryTypes } from 'vtex.product-summary-context'
+// @ts-expect-error
```
Comment (kaisermann) [inline react/hooks/useSimulation.ts:4]: Need an explanation

File: react/ProductSummaryImage.tsx [hunk 4]
```diff
@@ -271,7 +272,9 @@ function ProductImage({
 
   const [width, height] = [
     // fallsback to the other remaining value, if not defined
+    // @ts-expect-error
     parseFloat(responsiveWidth ?? responsiveHeight ?? 0),
+    // @ts-expect-error
     parseFloat(responsiveHeight ?? responsiveWidth ?? 0),
```
Comment (kaisermann) [inline react/ProductSummaryImage.tsx]: Need explanations :grin:

**Review by kaisermann — CHANGES_REQUESTED** [review]
Tests are not passing

**Review by kaisermann — APPROVED** [review]

File: react/utils/aspectRatioUtil.ts [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { changeImageUrlSize } from './normalize'
  * For example, turns "3:4" into 1.333, so the image height will be
  * 1.333 times its width.
  */
-const parseAspectRatio = (input) => {
+const parseAspectRatio = (input: string) => {
```
Comment (victorhmp) [inline react/utils/aspectRatioUtil.ts:7]:
```suggestion
const parseAspectRatio = (input: string | number) => {
```

File: react/ProductSummaryCustom.tsx [hunk 6]
```diff
@@ -1,66 +1,66 @@
...
+import ProductPriceSimulationWrapper from './components/ProductPriceSimulationWrapper'
+
+const {
+  ProductSummaryProvider,
+  useProductSummaryDispatch,
+  useProductSummary,
+} = ProductSummaryContext
 
 const PRODUCT_SUMMARY_MAX_WIDTH = 300
 const CSS_HANDLES = ['container', 'containerNormal', 'element', 'clearLink']
```
Comment (victorhmp) [inline react/ProductSummaryCustom.tsx:24]:
```suggestion
const CSS_HANDLES = ['container', 'containerNormal', 'element', 'clearLink'] as const
```

File: react/ProductSummaryCustom.tsx [hunk 7]
```diff
@@ -162,44 +161,49 @@ const ProductSummaryCustom = ({
...
       </ProductContextProvider>
-    </ProductSummaryContext.Provider>
+    </LocalProductSummaryContext.Provider>
   )
 }
 
-ProductSummaryCustom.propTypes = {
-  /** Product that owns the informations */
-  product: productShape,
+interface Props {
```
Comment (victorhmp) [inline react/ProductSummaryCustom.tsx:168]: Why not also add `classes` in this component? I've just noticed the Wrapper below, you would need to include it there too :)

File: react/ProductSummarySKUSelector.tsx [hunk 8]
```diff
@@ -1,24 +1,28 @@
...
+const { useProductSummary, useProductSummaryDispatch } = ProductSummaryContext
+
+const CSS_HANDLES = ['SKUSelectorContainer'] as const
+
+interface Props {
+  classes: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
+}
+
+function ProductSummarySKUSelector(props: Props) {
+  const { handles } = useCssHandles(CSS_HANDLES)
```
Comment (victorhmp) [inline react/ProductSummarySKUSelector.tsx:17]:
```suggestion
  const { handles } = useCssHandles(CSS_HANDLES, { classes })
```

**Comment by mateuspontes** [comment]
Great job, @klzns! All the commits are self-explanatory ✅

**Review by victorhmp — CHANGES_REQUESTED** [review]

**Review by victorhmp — APPROVED** [review]

File: react/ProductSummarySKUSelector.tsx [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+  variationsSpacing?: number
+  showVariationsErrorMessage?: boolean
+  initialSelection?: 'complete' | 'image' | 'empty'
+  displayMode?: ResponsiveValuesTypes.ResponsiveValue<
+    'select' | 'default' | 'slider'
+  >
+  sliderDisplayThreshold?: number
+  sliderArrowSize?: number
+  sliderItemsPerPage?: ResponsiveValuesTypes.ResponsiveValue<number>
+  classes: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```
Comment (icazevedo) [inline react/ProductSummarySKUSelector.tsx:56]:
```suggestion
  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```

File: react/utils/aspectRatioUtil.ts [hunk 10]
```diff
@@ -38,12 +38,17 @@ const parseAspectRatio = (input) => {
   return null
 }
 
-export const imageUrl = (src, size, maxSize, aspectRatio) => {
+export const imageUrl = (
+  src: string,
+  size: number,
+  maxSize: number,
+  aspectRatio: string
```
Comment (icazevedo) [inline react/utils/aspectRatioUtil.ts:45]:
```suggestion
  aspectRatio: string | number
```

File: react/ProductSummarySKUName.tsx [hunk 11]
```diff
@@ -1,12 +1,19 @@
...
+import type { CssHandlesTypes } from 'vtex.css-handles'
+import { ProductSummaryContext } from 'vtex.product-summary-context'
+
+const { useProductSummary } = ProductSummaryContext
 
 const CSS_HANDLES = ['skuNameContainer'] as const
 
-const ProductSummarySKUName = () => {
+interface Props {
+  classes: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```
Comment (icazevedo) [inline react/ProductSummarySKUName.tsx:11]:
```suggestion
  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```

File: react/ProductSummaryDescription.tsx [hunk 12]
```diff
@@ -1,18 +1,25 @@
...
+import { ProductSummaryContext } from 'vtex.product-summary-context'
+
+const { useProductSummary } = ProductSummaryContext
 
 const MAX_SIZE_DESCRIPTION = 120
 const CSS_HANDLES = ['description'] as const
 
-function ProductSummaryDescription() {
+interface Props {
+  classes: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```
Comment (icazevedo) [inline react/ProductSummaryDescription.tsx:13]:
```suggestion
  classes?: CssHandlesTypes.CustomClasses<typeof CSS_HANDLES>
```

File: react/ProductSummaryBuyButton.tsx [hunk 13]
```diff
@@ -1,44 +1,60 @@
...
-  const handles = useCssHandles(CSS_HANDLES)
+}: Props) {
+  const {
+    hints: { mobile },
+  } = useRuntime()
+
+  const { product, selectedItem, selectedQuantity } =
+    ProductSummaryContext.useProductSummary() ?? {}
+
+  const { handles } = useCssHandles(CSS_HANDLES)
```
Comment (icazevedo) [inline react/ProductSummaryBuyButton.tsx:57]: Shouldn't this component have `classes` too?

File: react/ProductSummaryCustom.tsx [hunk 14]
```diff
@@ -162,44 +168,52 @@ const ProductSummaryCustom = ({
...
+  product: ProductSummaryTypes.Product
   /** Function that is executed when a product is clicked */
-  actionOnClick: PropTypes.func,
-  children: PropTypes.node,
-  containerRef: PropTypes.oneOfType([
```
Comment (icazevedo) [inline react/ProductSummaryCustom.tsx:175]: What did this prop do and why could it be removed?

File: react/ProductSummaryImage.tsx [hunk 15]
```diff
@@ -0,0 +1,409 @@
...
+
+const MAX_SIZE = 500
+const DEFAULT_SIZE = 300
+
+type GetImageSrcParams = {
+  src: string
+  width: number
+  height: number
+  dpi: number
+  aspectRatio?: string
```
Comment (icazevedo) [inline react/ProductSummaryImage.tsx:36]:
```suggestion
  aspectRatio?: string | number
```
To match the aspect ratio util

File: react/ProductSummaryImage.tsx [hunk 16]
```diff
@@ -0,0 +1,409 @@
...
+    return imageUrl(src, DEFAULT_SIZE, MAX_SIZE, aspectRatio)
+  }
+
+  return src
+}
+
+type GetStyleParams = {
+  width: number
+  height: number
+  aspectRatio?: string
```
Comment (icazevedo) [inline react/ProductSummaryImage.tsx:60]:
```suggestion
  aspectRatio?: string | number
```
Same as above

File: react/ProductSummaryImage.tsx [hunk 17]
```diff
@@ -0,0 +1,409 @@
...
+      label={badgeText}
+    >
+      {children}
+    </DiscountBadge>
+  )
+}
+
+function findImageByLabel(
+  images: Array<{ imageLabel: string; imageUrl: string }>,
+  selectedLabel: string
```
Comment (icazevedo) [inline react/ProductSummaryImage.tsx:139]:
```suggestion
  selectedLabel: string | undefined
```

File: react/ProductSummaryImage.tsx [hunk 18]
```diff
@@ -0,0 +1,409 @@
...
+}
+
+interface ImageProps {
+  src: string
+  width: number
+  height: number
+  onError: () => void
+  alt: string
+  className: string
+  aspectRatio?: string
```
Comment (icazevedo) [inline react/ProductSummaryImage.tsx:155]:
```suggestion
  aspectRatio?: string | number
```
Same as above

File: react/ProductSummaryImage.tsx [hunk 19]
```diff
@@ -0,0 +1,409 @@
...
+  hoverImageLabel?: string
+  /**
+   * Defines if the collection badges are shown
+   * @default false
+   */
+  showCollections?: boolean
+  placeholder?: string
+  width?: ResponsiveValuesTypes.ResponsiveValue<number | string>
+  height?: ResponsiveValuesTypes.ResponsiveValue<number | string>
+  aspectRatio?: ResponsiveValuesTypes.ResponsiveValue<string>
```
Comment (icazevedo) [inline react/ProductSummaryImage.tsx:224]:
```suggestion
  aspectRatio?: ResponsiveValuesTypes.ResponsiveValue<string | number>
```

Same as above

File: react/ProductSummaryImage.tsx [hunk 20]
```diff
@@ -0,0 +1,409 @@
+/* eslint-disable react/prop-types */
```
Comment (icazevedo) [inline react/ProductSummaryImage.tsx:1]: Is this still needed?

File: react/ProductSummaryBuyButton.tsx [hunk 21]
```diff
@@ -1,44 +1,60 @@
 import React from 'react'
-import PropTypes from 'prop-types'
-import BuyButton from 'vtex.store-components/BuyButton'
-import { withRuntimeContext } from 'vtex.render-runtime'
+import { BuyButton } from 'vtex.store-components'
```
Comment (icazevedo) [inline react/ProductSummaryBuyButton.tsx:2]: This component still uses `BuyButton`, but we deprecated it on `store-components`. Should we refactor ProductSummaryBuyButton to use `add-to-cart-button` in the future?

**Review by icazevedo — CHANGES_REQUESTED** [review]

File: react/ProductSummaryBuyButton.tsx (hunk 13 above)
Comment (klzns) [reply react/ProductSummaryBuyButton.tsx:57]: It's deprecated

File: react/ProductSummaryCustom.tsx (hunk 14 above)
Comment (klzns) [reply react/ProductSummaryCustom.tsx:175]: It was not being used 🤷‍♂️

Comment (icazevedo) [reply react/ProductSummaryCustom.tsx:175]: It was sent to the context via the `...rest`, wasn't it?

Comment (icazevedo) [reply react/ProductSummaryCustom.tsx:175]: I've found a changelog entry saying it was removed on version 2.51.2. Now I'm convinced it's safe 😄

File: react/ProductSummaryImage.tsx (hunk 20 above)
Comment (klzns) [reply react/ProductSummaryImage.tsx:1]: good catch

File: react/ProductSummaryBuyButton.tsx (hunk 21 above)
Comment (klzns) [reply react/ProductSummaryBuyButton.tsx:2]: That's the pair, we should deprecate this component too.

**Review by icazevedo — APPROVED** [review]

**Pull request merged**